### PR TITLE
Add whitelistforcerelay to control forced relaying. [#7099 redux]

### PR DESCRIPTION
**Pull request by gmaxwell**
Also renames whitelistalwaysrelay.

Nodes relay all transactions from whitelisted peers, this
 gets in the way of some useful reasons for whitelisting
 peers-- for example, bypassing bandwidth limitations.

The purpose of this forced relaying is for specialized gateway
 applications where a node is being used as a P2P connection
 filter and multiplexer, but where you don't want it getting
 in the way of (re-)broadcast.

This change makes it configurable with whitelistforcerelay.

### CODE REVIEW

**Comment by laanwj** [comment]
utACK 325c725

File: src/init.cpp [hunk 1]
```diff
@@ -752,13 +753,19 @@ void InitParameterInteraction()
 
     // disable walletbroadcast and whitelistalwaysrelay in blocksonly mode
```
Comment (MarcoFalke) [inline src/init.cpp]: You'd need to make the rename globally; Also, I don't like that this just ignores    `-whitelistalwaysrelay` set in old `.conf` files.

**Comment by MarcoFalke** [comment]
Concept ACK

File: src/init.cpp (hunk 1 above)
Comment (laanwj) [inline src/init.cpp]: Valid point, although how old can these conf files be, given that `whitelistalwaysrelay` was introduced in #6993 only two months ago, so isn't in any releases.

Comment (MarcoFalke) [inline src/init.cpp]: I guess there are three options:

* Silently accept both variations
* Ignore
* Add it to the blacklist https://github.com/gmaxwell/bitcoin/blob/325c725fb6205e38142914acb9ed1733d8482d46/src/init.cpp#L892-L903

**Comment by petertodd** [comment]
utACK 325c725fb6205e38142914acb9ed1733d8482d46

**Comment by laanwj** [comment]
Travis is failing, but I cannot quite see why.

**Comment by gmaxwell** [comment]
I think it started failing when I added the requested argument rejection. Does a test run with it? git grep yields nothing.

**Comment by MarcoFalke** [comment]
utACK  89d113e. Thanks for sticking with this, I like the confs.

**Pull request merged**